apps/crawler/src/subscribers/collections-subscriber.service.ts

```diff
@@ -151,6 +143,37 @@ export class CollectionsSubscriberService implements ISubscriberService {
     return result;
   }
 
+  private getSchemaValuesFromProperties(
+    properties: CollectionProperty[],
+  ): UniqueCollectionSchemaDecoded {
+    const result = {
+      schemaName: '_properties_',
+      schemaVersion: '',
+      attributesSchemaVersion: '',
+      coverPicture: {},
+    } as ParsedSchemaFields;
+
+    Object.values(properties).reduce((acc, curr) => {
+      let { key, value } = curr;
+      try {
+        value = JSON.parse(value);
+      } catch (_) {
+        // I should try
+      }
+
+      let obj = acc as { coverPicture?: object };
+      if (key.startsWith('coverPicture.')) {
+        key = key.replace('coverPicture.', '');
+        obj = acc['coverPicture'];
+      }
+
+      obj[key] = value;
```

Comment:
Looks like acc chenging by change in obj variable. Why we are using obj variable?

Reply:
Maybe
acc = { acc..., acc['coverPicture'] }
delete acc['coverPicture']

or Object.assing?

Reply:
Agree, it looks a little bit weird. The idea is to fill up the main object `acc` and it's sub-object by key `coverPicture` in case we get `curr.key` like` 'coverPicture.SUB_KEY'.`

Reply:
Removed `obj` variable usage.